subsys/bluetooth/mesh/prov_device.c

```diff
@@ -291,16 +343,21 @@ static void prov_dh_key_gen(void)
 
 	remote_pk = &bt_mesh_prov_link.conf_inputs[17];
 
-	/* Copy remote key in little-endian for bt_dh_key_gen().
-	 * X and Y halves are swapped independently. The bt_dh_key_gen()
-	 * will also take care of validating the remote public key.
-	 */
-	sys_memcpy_swap(remote_pk_le, remote_pk, 32);
-	sys_memcpy_swap(&remote_pk_le[32], &remote_pk[32], 32);
-
-	if (bt_dh_key_gen(remote_pk_le, prov_dh_key_cb)) {
-		BT_ERR("Failed to generate DHKey");
-		prov_fail(PROV_ERR_UNEXP_ERR);
+	if (atomic_test_bit(bt_mesh_prov_link.flags, OOB_PUB_KEY) &&
+	    atomic_test_bit(bt_mesh_prov_link.flags, OOB_PUB_KEY_STATIC)) {
+		static_dh_key_gen();
+	} else {
+		/* Copy remote key in little-endian for bt_dh_key_gen().
+		 * X and Y halves are swapped independently. The bt_dh_key_gen()
+		 * will also take care of validating the remote public key.
+		 */
+		sys_memcpy_swap(remote_pk_le, remote_pk, 32);
+		sys_memcpy_swap(&remote_pk_le[32], &remote_pk[32], 32);
+
+		if (bt_dh_key_gen(remote_pk_le, prov_dh_key_cb)) {
```

Comment:
Is it necessary to use the existing structure and add specific functions so that the bt controller can calculate the given private/public key ?